src/App.tsx

```diff
@@ -1,61 +1,163 @@
+import React, { useState } from 'react';
 import './App.scss';
-
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo, User } from './types/types';
 
 export const App = () => {
+  // Helper function to get a user by their ID
+  const getUserById = (userId: number): User | null => {
+    return usersFromServer.find(user => user.id === userId) || null;
+  };
+
+  // Initial processing of todos to include the user object
+  const prepareInitialTodos = (): Todo[] => {
+    return todosFromServer.map(todo => ({
+      ...todo,
+      user: getUserById(todo.userId),
+    }));
+  };
+
+  // State initialization
+  const [userList] = useState(usersFromServer);
+  const [todosList, setTodosList] = useState(prepareInitialTodos());
+  const [titleInput, setTitleInput] = useState('');
+  const [userSelected, setUserSelected] = useState('');
```

Comment:
The `userSelected` state is initialized as a string, but it is used as a number in `createNewTodo`. Consider initializing it as a number or converting it appropriately when needed.